File: api/Dtos/Subscription/SubscriptionRes.cs
using api.Entities;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using api.Dtos.Plan;
using api.Dtos.User;

namespace api.Dtos.Subscription
{
    public class SubscriptionRes
    {
        public int SubscriptionId { get; set; }
        public int? PlanId { get; set; }
        public PlanRes? Plan { get; set; }
        public string? UserId { get; set; }
        public UserRes? User { get; set; }
        public DateTime ActiveFrom { get; set; } = DateTime.MinValue;
        public DateTime ActiveTo { get; set; } = DateTime.MinValue;
        public bool Status { get; set; }
    }
}

File: web/src/dtos/Subscription/SubscriptionRes.ts
import { Common } from "../../utility";
import { PlanRes } from "../Plan";
import { UserRes } from "../User";

export default class SubscriptionRes {
  subscriptionId?: string = "";
  planId?: string = "";
  plan?: PlanRes;
  userId?: string = "";
  user?: UserRes;
  activeFrom?: string = "";
  activeTo?: string = "";
  status?: boolean = false;
}
